api/src/main/java/com/datastrato/gravitino/authorization/AuthorizationRole.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+/** Authorization Role interface. */
+public interface AuthorizationRole {
+  /**
+   * Create a role.
+   *
+   * @param name The role's name.
+   * @return The created role object.
+   * @throws UnsupportedOperationException If the authorization does not support to create a role.
+   */
+  Role createRole(String name) throws UnsupportedOperationException;
+
+  /**
+   * Drop a role.
+   *
+   * @param role to be deleted.
+   * @return True if the delete operation success; False otherwise.
+   * @throws UnsupportedOperationException If the authorization does not support to drop a role.
+   */
+  Boolean dropRole(Role role) throws UnsupportedOperationException;
```

Comment:
I think using `name` string to drop a role is enough, WDYT?

Reply:
Only has role name is not enough.
Because when we delete a role in the Ranger, We need through `role.securableObjects` to find Ranger Policy and delete it.